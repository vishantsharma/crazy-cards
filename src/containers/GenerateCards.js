import React from "react";
import CheckBox from "../components/ui/CheckBox";
import Card from '../components/ui/Card';
import './GenerateCards.css';

const GenerateCards = ({ data, selected, handleClick }) => {
  return (
    <>
      <Card>
        <h2 className="heading">{data.card_name}</h2>
        <p>{'Apr:'} <span><strong>{`${data.apr}%`}</strong></span></p>
        <p>{'Balance Transfer Offer Duration:'} <span><strong>{`${data.balance_transfer_offer} months`}</strong></span></p> 
        <p>{'Purchase Offer Duration:'}  <span><strong>{`${data.purchase_offer} months`}</strong></span></p>
        <p>{'Credit Available:'} <span><strong>{`Â£${data.available_credit}`}</strong></span></p>
        <CheckBox
          name={`card-${data.id}`}
          label=" Select this Card"
          checked={selected}
          onClick={() => handleClick(data)}
        />
      </Card>
    </>
  );
};

export default GenerateCards;
