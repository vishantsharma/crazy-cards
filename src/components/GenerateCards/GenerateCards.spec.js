import React from 'react';
import TestRenderer from 'react-test-renderer';
import GenerateCards from './GenerateCards';

describe('GenerateCards', () => {
    let props = {
        data: '',
        selected: '',
        handleClick: jest.fn().mockReturnValue()
    }
  it('should render', () => {
    const component = TestRenderer.create(<GenerateCards {...props} />)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})