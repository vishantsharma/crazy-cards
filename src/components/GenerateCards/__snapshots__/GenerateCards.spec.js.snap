// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateCards should render 1`] = `
<div
  className="card"
>
  <h2
    className="heading"
  />
  <p>
    Apr:
     
    <span>
      <strong>
        undefined%
      </strong>
    </span>
  </p>
  <p>
    Balance Transfer Offer Duration:
     
    <span>
      <strong>
        undefined months
      </strong>
    </span>
  </p>
  <p>
    Purchase Offer Duration:
      
    <span>
      <strong>
        undefined months
      </strong>
    </span>
  </p>
  <p>
    Credit Available:
     
    <span>
      <strong>
        Â£undefined
      </strong>
    </span>
  </p>
  <input
    checked=""
    id="card-undefined"
    name="card-undefined"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="card-undefined"
  >
     Select this Card
  </label>
</div>
`;
